### PR TITLE
Add optional docker image with dependency sources

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Necessary documentation (if appropriate) have been added / updated
- [X] DCO signoffs have been added to all commits, including this PR

#### PR type
- [ ] Bugfix
- [X] Feature
- [ ] Other... Please describe:

#### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

#### Does this PR do something the person installing Zowe should know about?
This PR does a few things
* Updates the README of the server-bundle container
* Updates the dependency image to always run an update when building (to get the latest versions) and to use a newer version of nodejs
* Adds a new dockerfile to add sourcecode of dependencies into the server-bundle image, but as an optional alternative to the server-bundle image so you can have it with or without sources. This is needed for license compliance, but may be desirable to users too.

### CODE REVIEW

File: containers/server-bundle/Dockerfile.nodejava.amd64 [hunk 1]
```diff
@@ -3,20 +3,15 @@ FROM openjdk:8-jdk-slim
 RUN apt-get update && apt-get install -y --no-install-recommends \
 		ca-certificates curl wget \
     groff iproute2 moreutils pax && \
-	  rm -rf /var/lib/apt/lists/*
+    apt-get dist-upgrade -y
```
Comment (1000TurquoisePogs) [inline containers/server-bundle/Dockerfile.nodejava.amd64:6]: This should protect against docker images being a little outdated

File: containers/server-bundle/Dockerfile.nodejava.amd64 [hunk 2]
```diff
@@ -42,7 +37,7 @@ RUN set -ex \
     gpg --keyserver pool.sks-keyservers.net --recv-keys "$key" ; \
   done
 
-ENV NODE_VERSION 12.18.4
+ENV NODE_VERSION 12.20.1
```
Comment (1000TurquoisePogs) [inline containers/server-bundle/Dockerfile.nodejava.amd64:40]: We could go to v14, but when?

File: containers/utils/download-zowe-build.sh [hunk 3]
```diff
@@ -16,10 +16,14 @@ cd /root/zowe
 
 if [ "$ZOWE_BUILD" = "latest" ]; then
   echo "Downloading most recent release ..."
+  apt-get update && apt-get install -y wget --no-install-recommends
   wget -q "https://zowe.jfrog.io/zowe/libs-release-local/org/zowe/[RELEASE]/zowe-[RELEASE].pax" -O "$ZOWE_TARGET_NAME"
+  apt-get purge -y wget
```
Comment (1000TurquoisePogs) [inline containers/utils/download-zowe-build.sh:21]: we remove wget that we use during build time, but maybe its something we should keep in the image for use by plugins?

File: containers/utils/get-deb-sources.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT
```
Comment (swinslow) [inline containers/utils/get-deb-sources.sh:2]: As the original author of an earlier version of this script (see https://gitlab.com/swinslow/contcompl/-/blob/master/take2/get-sources.sh) -- if you would like to mark this as the following instead, that would be fine with me:

```
SPDX-License-Identifier: EPL-2.0 OR MIT
```

Just offering that as an option, for better alignment with Zowe's own license  :)

**Review by jackjia-ibm — APPROVED** [review]
Looks good to me, no obvious issues identified, source builds are showing in artifactory. Will see how release goes.

**Pull request merged**